03week/towersOfHanoi.js

```diff
@@ -7,48 +7,116 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+// ORIGINAL
 let stacks = {
   a: [4, 3, 2, 1],
   b: [],
   c: []
 };
 
+// TESTER
+// let stacks = {
+//   a: [],
+//   b: [1],
+//   c: [4, 3, 2]
+// };if
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+function movePiece(startStack, endStack) {
+  // take the top number of the startstack and move it to the endstack
 
+  let tempStack = stacks[startStack].pop();
+  stacks[endStack].push(tempStack);
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
```

Comment:
This function works and is a great demonstration of your knowledge of javascript methods! But it also is crazy convoluted for such a simple test. Since you're only checking for a few things, I would recommend something like
```
if (startStack.length) {
  if (!endStack.length || startStack[startStack.length - 1] < endStack[endStack.length - 1]) {
    return true;
  }
}```